rocAL/source/readers/file_source_reader.cpp

```diff
@@ -202,40 +253,39 @@ Reader::Status FileSourceReader::open_folder() {
     if ((_src_dir = opendir(_folder_path.c_str())) == nullptr)
         THROW("FileReader ShardID [" + TOSTR(_shard_id) + "] ERROR: Failed opening the directory at " + _folder_path);
 
-    while ((_entity = readdir(_src_dir)) != nullptr) {
-        if (_entity->d_type != DT_REG)
+    std::vector<filesys::path> files_in_directory;
+    std::copy(filesys::directory_iterator(filesys::path(_folder_path)), filesys::directory_iterator(), std::back_inserter(files_in_directory));
```

Comment:
What us this copy for?

Reply:
Copying the directory_iterator contents to a vector for sorting